commit 26f028467df2d3699cdf43a0a46daf90ebdd4eb3
Author: Mike Gerdts <mike.gerdts@joyent.com>
Date:   2019-08-01T20:07:48+00:00 (10 weeks ago)
    
    OS-7916 bash completions for vmadm snapshot commands do not support bhyve
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
    Reviewed by: John Levon <john.levon@joyent.com>
    Approved by: John Levon <john.levon@joyent.com>

diff --git a/src/vm/etc/vmadm.completion b/src/vm/etc/vmadm.completion
index 0e9fbc78..89871e13 100644
--- a/src/vm/etc/vmadm.completion
+++ b/src/vm/etc/vmadm.completion
@@ -41,9 +41,8 @@ _vmadm()
         || ${prev} == 'delete-snapshot'
         || ${prev} == 'rollback-snapshot' ]]; then
 
-        # OS only commands
-        vms_uuids=$(zoneadm list -cp | grep -v ':global:' \
-            | grep -v ":[bh]*[ky]v[em]:excl:" | cut -d':' -f5 | sort | uniq)
+        vms_uuids=$(zoneadm list -cp |
+	    nawk -F: '$2 != "global" && $6 != "kvm" { print $5 }' | sort -u)
         COMPREPLY=( $(compgen -W "${vms_uuids}" -- ${cur}) )
 
     elif [[ ${prev} == 'validate' ]]; then
